src/components/accounts/recovery_setup/SetupRecoveryMethod.js

```diff
@@ -40,27 +41,45 @@ class SetupRecoveryMethod extends Component {
         success: false,
         emailInvalid: false,
         phoneInvalid: false,
-        activeMethods: []
+        activeMethods: [],
+        hasFetchedMethods: false
     }
 
     componentDidMount() {
-        const { loadRecoveryMethods, accountId, router, recoveryMethods } = this.props;
+        const { router, getAccessKeys } = this.props;
         const { method } = router.location;
 
+        getAccessKeys()
+
         if (method) {
             this.setState({ option: method });
         }
 
-        if (recoveryMethods[accountId]) {
-            const confirmed = recoveryMethods[accountId].filter(method => method.confirmed)
+        this.setRecoveryMethods()
+    }
+
+    setRecoveryMethods = () => {
+        if (this.props.recoveryMethods[this.props.accountId]) {
+            const confirmed = this.props.recoveryMethods[this.props.accountId].filter(method => method.confirmed)
             this.setState({ activeMethods: confirmed.map(method => method.kind) });
         } else {
-            loadRecoveryMethods(accountId)
-                .then((payload) => {
-                    const confirmed = payload.data.filter(method => method.confirmed);
-                    this.setState({ activeMethods: confirmed.map(method => method.kind) });
-                })
+            if (!this.state.hasFetchedMethods) {
+                this.getMethods()
+            }
+        }
+    }
+
+    getMethods = async () => {
+
+        try {
+            await this.props.loadRecoveryMethods()
+        } catch(e) {
+            console.log(e)
```

Comment:
why is error ignored?

Reply:
@Patrick1904 I think referring to line 139-141
```
const { error } = await setupRecoveryMessage(accountId, this.method, securityCode)

        if (error) {
            return;
        }
```

Reply:
The only purpose of this is to check which methods are already active so that we can disable the option to enable them. Does it hurt to continue 'updating' which methods are active?

Reply:
@Patrick1904 not sure what do you mean. Error is ignored too check which methods are already active?